applications/portal/backend/api/views/rest/experiment.py

```diff
@@ -217,3 +224,73 @@ def download_populations(self, request, pk=None, active_populations=None):
 
     def perform_create(self, serializer):
         experiment = serializer.save(owner=self.request.user)
+
+    def check_if_user_is_owner(self, kwargs_experiment_id):
+        try:
+            user = get_current_user()
+            experiment = Experiment.objects.get(id=kwargs_experiment_id)
+            if user.id != experiment.owner.id:
+                return Response(status=status.HTTP_403_FORBIDDEN, data={'detail': 'You are not the owner of the experiment'})
+        except Experiment.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND, data={'detail': 'Experiment does not exist'})
+        
+    def check_if_subpopulation_exists(self, subpopulation_id, kwargs_experiment_id):
+        try:
+            sub_population_object = Population.objects.get(id=subpopulation_id)
+            is_residential = sub_population_object.experiment is None
+            if is_residential:
+                return Response(status=status.HTTP_403_FORBIDDEN, data={'detail': 'You cannot change the name of a residential population'})
+            if int(sub_population_object.experiment.id) != int(kwargs_experiment_id):
+                return Response(status=status.HTTP_404_NOT_FOUND, data={'detail': f'Subpopulation with id {subpopulation_id} does not exist in this experiment'})
+        except Population.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND, data={'detail': f'Subpopulation with id {subpopulation_id} does not exist'})
+        return sub_population_object
+    
+
+    @action(
+        detail=True, 
+        methods=['patch'], 
+        parser_classes=(JSONParser,),
+        url_name="rename_population"
+    )
+    def rename_population(self, request, **kwargs):
+        kwargs_experiment_id = kwargs['pk']
+        is_user_owner = self.check_if_user_is_owner(kwargs_experiment_id)
+        request_data = json.loads(request.body)
+        sub_populations = []
+        try:
+            for change in request_data['change']:
```

Comment:
@D-GopalKrishna why can't we use the "plain" patch of a population REST call?
I understand you want to change the names of a set of populations but you can also make live easier by sending the population records to a path that supports a list of populations (this is a new method) OR just loop in the frontend and call per population the patch method

Reply:
The idea behind it was that = if we loop in Frontend, and if a population has let's say 20 subpopulation, it would make 20 API calls, and it might be easier to do that in just one API call.

Reply:
I understand but the question is is this a validate readson for the creation of a new method that can contain bugs... imo no it isn't a valid reason